File: solutions/cpp/0929.cpp
class Solution {
 public:
  int numUniqueEmails(vector<string>& emails) {
    unordered_set<string> seen;

    for (const string& email : emails) {
      string local;
      for (char c : email) {
        if (c == '+' || c == '@') break;
        if (c == '.') continue;
        local += c;
      }
      string atDomain = email.substr(email.find('@'));
      seen.insert(local + atDomain);
    }

    return seen.size();
  }
};

File: solutions/java/0929.java
class Solution {
  public int numUniqueEmails(String[] emails) {
    Set<String> seen = new HashSet<>();

    for (final String email : emails) {
      String[] parts = email.split("@");
      String[] local = parts[0].split("\\+");
      seen.add(local[0].replace(".", "") + "@" + parts[1]);
    }

    return seen.size();
  }
}